src/main/java/io/boomerang/eventing/nats/jetstream/PubTransmitter.java

```diff
@@ -80,32 +87,114 @@ public void publish(String subject, String message)
 
     // Get NATS connection
     Connection connection = connectionPrimer.getActiveConnection();
-
     if (connection == null) {
+      storeMessage(subject, message);
       throw new NoNatsConnectionException("No connection to the NATS server!");
```

Comment:
If there is no connection to the NATS server, it should not throw an exception. This must be transparent to the developer and the retry mechanism will start working automatically.